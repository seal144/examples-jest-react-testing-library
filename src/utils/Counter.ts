class Counter {
  value: number;
  constructor() {
    this.value = 0;
  }
  increment(incrementValue?: number) {
    incrementValue ? (this.value += incrementValue) : (this.value += 1);
  }
  decrement(decrementValue?: number) {
    decrementValue ? (this.value -= decrementValue) : (this.value -= 1);
  }
  getValue() {
    return this.value;
  }
}

export default Counter;
