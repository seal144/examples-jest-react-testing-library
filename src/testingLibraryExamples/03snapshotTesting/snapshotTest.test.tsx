import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import Counter from '../../components/Counter';

test('matches snapshot after incrementing count', () => {
  const { asFragment } = render(<Counter />);

  const incrementButton = screen.getByRole('button');

  // Initial render
  expect(asFragment()).toMatchSnapshot();

  // Fire click event on the button
  fireEvent.click(incrementButton);

  // Snapshot after click event
  expect(asFragment()).toMatchSnapshot();
});
