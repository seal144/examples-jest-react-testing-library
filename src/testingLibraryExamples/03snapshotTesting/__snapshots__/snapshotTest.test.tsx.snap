// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot after incrementing count 1`] = `
<DocumentFragment>
  <div>
    <p>
      Count: 0
    </p>
    <button>
      Increment
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot after incrementing count 2`] = `
<DocumentFragment>
  <div>
    <p>
      Count: 1
    </p>
    <button>
      Increment
    </button>
  </div>
</DocumentFragment>
`;
